### DISCUSSION TITLE
How to prevent elements from reinitialized after DOM changed?

### DISCUSSION THREAD

**Discussion by teikun-86** [alpinejs/alpine#2198] [1. Help] [answered]
Hi, i'm trying to create a form where the `input` element is can added as it needed but i have a problem that the modal is always closed after the DOM changed, i have tried to add `x-ignore` to the modal element, but the modal element is not initialized. Anyone have solution for my problem?

**Comment by Tim-Wils** [comment]
Hi! Show us a minimal example of your problem in a codepen, so we can see it and help you with the problem 😄👍🏻

**Reply by teikun-86** [reply]
just fixed it by reinstalling alpine js via npm, i don't know why,
thanks for answering

**Comment by teikun-86** [comment, marked as answer]
Now after the modal problem is solved, i don't realize that the form is resetted when i add new input like this
https://codepen.io/azizsama/pen/RwZNMYa what's wrong with my code?

**Reply by Tim-Wils** [reply]
@azizsama You could do something like this: https://codepen.io/tim-wils/pen/qBXEoeE

**Reply by teikun-86** [reply]
@Tim-Wils thanks you! its work 😁
